tmp_file.h

```diff
@@ -0,0 +1,121 @@
+/*
+    tmp_file.h - write to and read from a temporary binary file
+    for fast storage plus added compression.
+    
+    Andrew Whitwham, August 2017
```

Comment:
This file needs a specific copyright statement and to indicate who the copyright holder is in addition to the author.